test/e2e/apimachinery/flowcontrol.go

```diff
@@ -0,0 +1,116 @@
+				ObjectMeta: metav1.ObjectMeta{
+					Name: testingPriorityLevelName,
+				},
+				Spec: flowcontrolv1alpha1.PriorityLevelConfigurationSpec{
+					Type: flowcontrolv1alpha1.PriorityLevelEnablementLimited,
+					Limited: &flowcontrolv1alpha1.LimitedPriorityLevelConfiguration{
+						AssuredConcurrencyShares: 1, // will have at minimum 1 concurrency share
+						LimitResponse: flowcontrolv1alpha1.LimitResponse{
+							Type: flowcontrolv1alpha1.LimitResponseTypeReject,
+						},
+					},
+				},
+			},
+			metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create a testing priority-level")
+
+		ginkgo.By("creating a testing flowschema")
+		createdFlowSchema, err := f.ClientSet.FlowcontrolV1alpha1().FlowSchemas().Create(
+			context.TODO(),
+			&flowcontrolv1alpha1.FlowSchema{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: testingFlowSchemaName,
+				},
+				Spec: flowcontrolv1alpha1.FlowSchemaSpec{
+					MatchingPrecedence: 1000, // a rather higher precedence to ensure it make effect
+					PriorityLevelConfiguration: flowcontrolv1alpha1.PriorityLevelConfigurationReference{
+						Name: testingPriorityLevelName,
+					},
+					DistinguisherMethod: &flowcontrolv1alpha1.FlowDistinguisherMethod{
+						Type: flowcontrolv1alpha1.FlowDistinguisherMethodByUserType,
+					},
+					Rules: []flowcontrolv1alpha1.PolicyRulesWithSubjects{
+						{
+							Subjects: []flowcontrolv1alpha1.Subject{
+								{
+									Kind: flowcontrolv1alpha1.SubjectKindUser,
+									User: &flowcontrolv1alpha1.UserSubject{
+										Name: matchingUsername,
+									},
+								},
+							},
+							NonResourceRules: []flowcontrolv1alpha1.NonResourcePolicyRule{
+								{
+									Verbs:           []string{flowcontrolv1alpha1.VerbAll},
+									NonResourceURLs: []string{flowcontrolv1alpha1.NonResourceAll},
+								},
+							},
+						},
+					},
+				},
+			},
+			metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create a testing flow-schema")
+
+		ginkgo.By("response headers should contain flow-schema/priority-level uid")
+		impersonatedClientConfig := rest.CopyConfig(f.ClientConfig())
+		impersonatedClientConfig.Impersonate.UserName = matchingUsername
+		roundTripper, err := rest.TransportFor(impersonatedClientConfig)
+		framework.ExpectNoError(err, "failed to create an impersonated client instance")
+
+		req, err := http.NewRequest(http.MethodGet, "/version", nil)
+		framework.ExpectNoError(err, "failed to craft a non-resource request against the testing flow-schema")
+		resp, err := roundTripper.RoundTrip(req)
+		if fsUID := resp.Header.Get(filters.ResponseHeaderMatchedFlowSchemaUID); fsUID != string(createdFlowSchema.ObjectMeta.UID) {
+			framework.Failf("uid mismatch from the testing flow-schema")
+		}
+		if plUID := resp.Header.Get(filters.ResponseHeaderMatchedPriorityLevelConfigurationUID); plUID != string(createdPriorityLevel.ObjectMeta.UID) {
+			framework.Failf("uid mismatch from the testing flow-schema")
+		}
```

Comment:
Also check that a non-matching request doesn't get the same headers?

Reply:
done